File: patterns/cs/FactoryMethod/WindowsDialog.cs
using System;

namespace Guru.Refactoring.Patterns.FactoryMethod.Example
{
    // Конкретные фабрики переопределяют фабричный метод и возвращают из него
    // свои собственные продукты.
    class WindowsDialog : Dialog
    {
        public override Button CreateButton()
        {
            return new WindowsButton();
        }
    }
}

File: patterns/java/src/refactoring_guru/patterns/factory_method/example/factory_method/example/factory/WindowsDialog.java
package refactoring_guru.patterns.factory_method.example.factory;

import refactoring_guru.patterns.factory_method.example.buttons.Button;
import refactoring_guru.patterns.factory_method.example.buttons.WindowsButton;

public class WindowsDialog extends Dialog {

    @Override
    public Button createButton() {
        return new WindowsButton();
    }
}
